### ISSUE TITLE
Ambiguous  element detected where none exists.

### ISSUE THREAD

**Issue by rustysys-dev** [idubinskiy/schematyper#13]
# Issue:
I am parsing a schema document based on JSON Schema version 7 and have run into the following error.

```shell
[yamada@localhost device-sender]$ schematyper data.json 
2018/12/21 17:08:21 Can't disabiguate: (#/properties/d/items #)
```
# Expectation:

I have verified the schema format should be correct, as it was largely generated in this format according to https://www.jsonschema.net, the only thing taht was added are title names, and pattern matching.

I would expect that the standard array type below which adheres to JSON schema v7 should be recognized with no ambiguities.

# Reference File:

```json
{
  "definitions": {},
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "http://example.com/system/interface/data.json",
  "type": "object",
  "title": "Stamp Message Schema",
  "required": [
    "t",
    "d"
  ],
  "properties": {
    "t": {
      "$id": "#/properties/t",
      "type": "string",
      "title": "msgType",
      "default": "",
      "examples": [
        "B"
      ],
      "pattern": "^([B])$"
    },
    "d": {
      "$id": "#/properties/d",
      "type": "array",
      "title": "dataArray",
      "items": {
        "$id": "#/properties/d/items",
        "type": "object",
        "title": "data",
        "required": [
          "64",
          "dt",
          "id"
        ],
        "properties": {
          "64": {
            "$id": "#/properties/d/items/properties/64",
            "type": "string",
            "title": "base64",
            "default": "",
            "examples": [
              "aGVsbG8gd29ybGQ="
            ],
            "pattern": "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$"
          },
          "dt": {
            "$id": "#/properties/d/items/properties/dt",
            "type": "string",
            "title": "dataType",
            "default": "",
            "examples": [
              "C"
            ],
            "pattern": "^(.*)$"
          },
          "id": {
            "$id": "#/properties/d/items/properties/id",
            "type": "integer",
            "title": "ID",
            "default": "",
            "examples": [
              1
            ]
          }
        }
      }
    }
  }
}
```

**Comment by idubinskiy** [comment]
@procyclinsur The issue is that the name of the `#/properties/d/items` type, which comes from the `title` property, conflicts with the name of the root type (`#`), which comes from the filename, so both would be `data`. I can probably find a better way to handle this, but there are a couple of workarounds: you can rename the file to something like `stamp.json`, or you can use the `--root-type` flag to override the name of the root type.

**Comment by rustysys-dev** [comment]
@idubinskiy Thank you for the fast response!

I ended up changing the title name, I suppose if this Issue is here people will be able to understand the workaround and adapt to it.. I don't see any reason to push for a fix to allow for this ambiguity to be considered ok.

Although I don't think its a good solution, we also found that schematyper passes this file if you were to change the type to "interface" without any changes elsewhere.

Thanks again!
